crates/hir_def/src/body/lower.rs

```diff
@@ -39,20 +40,33 @@ use crate::{
 
 use super::{diagnostics::BodyDiagnostic, ExprSource, PatSource};
 
-pub(crate) struct LowerCtx {
+pub struct LowerCtx {
     hygiene: Hygiene,
+    source_ast_id_map: Option<Arc<AstIdMap>>,
```

Comment:
The `LowerCtx` needs a `AstIdMap` so `TypeRef::from_ast` can construct the `TypeRef::Macro` variant which contains a `FileAstId<ast::MacroCall>`.